core/errors/error-match-error.ts

```diff
@@ -2,7 +2,7 @@ import { MatchError } from "./match-error";
 
 export class ErrorMatchError extends MatchError {
 
-   public constructor(actualError: Error,
+   public constructor(actualError: Error | undefined | null,
```

Comment:
I think I'd be happy with `Error | null` it was more the three unions seemed too much :)